### PR TITLE
Add GPDB 7 support

### PR DESCRIPTION
**Pull request by hilltracer**
Support for GPDB 7, based on PostgreSQL 12, has been integrated: 

- Update the CheckPoint structure and both PG_CONTROL_VERSION and
CATALOG_VERSION_NO to match [GPDB7 specifications](https://github.com/arenadata/gpdb/blob/46dc8aa0bca57ed56cd6babd57fa612a3d80ff29/src/include/catalog/pg_control.h#L40-L73). 
- Modify the postgres.yaml file and regenerate the corresponding autocode to
reflect these changes.
- Update the backup instructions. 
- Develop unit tests.

Backup and restore operations were successfully verified on Ubuntu 22.04 using 
version adb-7.1.0, adhering to updated [instruction](https://github.com/hilltracer/pgbackrest/blob/bbd8737c1ffc534f408f69db97d8deeaf8d4adb2/README_GPDB_en.md).

### CODE REVIEW

File: README_GPDB_en.md [hunk 1]
```diff
@@ -102,6 +108,7 @@ start-fast=y
 fork=GPDB
 EOF
 ```
+For Greenplum 7, the above command requires replacing the port numbers from 6000, 6002, 6003, 6004 to 7000, 7002, 7003, 7004, respectively.
```
Comment (bimboterminator1) [inline README_GPDB_en.md]: Maybe duplicate gpdb 6 commands for gpdb 7 using collapsible block?

<details>
  <summary>Example</summary>
  
  ### Heading
  1. Foo
  2. Bar
     * Baz
     * Qux

  ### Some Javascript
  ```js
  function logSomething(something) {
    console.log('Something', something);
  }
  ```
</details>

Comment (hilltracer) [reply README_GPDB_en.md]: Good idea! I've updated the instructions.

**Review by bimboterminator1 — DISMISSED** [review]

File: README_GPDB_en.md [hunk 2]
```diff
@@ -120,7 +165,7 @@ gpconfig -c archive_command -v "'PGOPTIONS=\"-c gp_session_role=utility\" /usr/l
 gpstop -ar
 ```
 
-- Install the gp_pitr extension
+- Install the gp_pitr extension (skip this step for GreenPlum 7)
```
Comment (red1452) [inline README_GPDB_en.md]: At the all places you wrote "Greenplum", but here you wrote "GreenPlum" (and at the same place in README_GPDB_ru.md).

Comment (hilltracer) [reply README_GPDB_en.md]: Thanks! I've fixed it.

**Review by red1452 — APPROVED** [review]

**Review by bimboterminator1 — APPROVED** [review]

**Pull request merged**